src/hooks/products/useProductInfo.jsx

```diff
@@ -0,0 +1,161 @@
+import {useEffect, useState} from "react";
+import errorNotif from "../../components/ErrorNotif";
+import useAxiosPrivate from "../useAxiosPrivate";
+import {showNotification} from "@mantine/notifications";
+import {IconCheck} from "@tabler/icons-react";
+
+/*
+Permet de récupérer la liste des produits d'une entité
+
+Paramètres :
+    productId : id du produit dont on veut les informations
+
+Retours :
+    isLoading : bool qui donne l'état de chargement des informations
+    isSending : bool qui donne l'état de l'envoi d'informations
+    retrieveProducts : Fonction qui force l'actualisation des informations du produit
+    updateProduct : Fonction pour mettre à jour les infos d'un produit
+    deleteProduct : Fonction pour supprimer un produit
+    productInfo :
+        {
+            "id": id du produit (int),
+            "raccourci": "raccourci pour trouver le produit",
+            "nom": "Nom du produit",
+            "prix": prix du produit en euros (float),
+            "entite": entité auquel le produit appartient,
+        }
+ */
+
+export function useProductInfo(productId){
+
+    const axiosPrivate = useAxiosPrivate()
+
+    const [isLoading, setLoading] = useState(true)
+    const [isSending, setSending] = useState(false)
+    const [productInfo, setProductInfo] = useState({})
+
+    const addProduct = async (productInfo)=>{
```

Comment:
toutes ces fonctions sont redéclarées à chaque re-render. lorsqu'on les met dans un useeffect, elles ne sont donc pas égales (Object.is)
Il faut les mettre dans un useCallback
```suggestion
    const addProduct = useCallback(async (productInfo)=>{
    ...
    }, [dependency array à faire en conséquence]);
```